pages/home.py

```diff
@@ -35,69 +37,188 @@ def get_countries(economies: Economies) -> list[str]:
-    df = (
-        dfc_imf_dot
-        .query(f"`Country Name` == '{country}'")['Counterpart Country Name']
-        .unique())
+    "Get counterpart country list based on country"
+    df = dfc_imf_dot.query(f"`Country Name` == '{country}'")[
+        "Counterpart Country Name"
+    ].unique()
     sorted_list = np.sort(df).tolist()
     if economies is Economies.COUNTRY:
         sorted_list = [x for x in sorted_list if x not in IMF_ECON_GROUPS]
     if economies is Economies.ECONOMIC_GROUP:
         sorted_list = [x for x in sorted_list if x in IMF_ECON_GROUPS]
-    return ['All'] + list(sorted_list)
+    return ["All"] + list(sorted_list)
+
+
+def get_top_n_counterpart_countries(
+    economies: Economies,
+    country: str,
+    top_n: int = 10,
+    year: int = 2021,
+    trade_type: str = "Export",
+) -> list[str]:
+    "Get the top n counterpart countries, ranked by trade volume"
+    df = dfc_imf_dot[
+        (dfc_imf_dot["Country Name"] == country)
+        & (dfc_imf_dot["Indicator Name"] == trade_type)
+    ]
+    trade_volume = (
+        df.groupby(["Counterpart Country Name"])[str(year)]
+        .sum()
+        .sort_values(ascending=False)
+        .reset_index()
+    )
+    top_n_counters = trade_volume["Counterpart Country Name"].tolist()
+
+    if economies is Economies.COUNTRY:
+        top_n_counters = [x for x in top_n_counters if x not in IMF_ECON_GROUPS]
+    if economies is Economies.ECONOMIC_GROUP:
+        top_n_counters = [x for x in top_n_counters if x in IMF_ECON_GROUPS]
+
+    top_n_counters = top_n_counters[:top_n]
+
+    return top_n_counters
 
 
 @st.experimental_memo
 def get_years(country, counterpart_country):
-    'Get non nan years'
-    years = [x for x in dfc_imf_dot.columns if bool(re.match(r'^\d{4}$', x))]
-    if counterpart_country == 'All':
+    "Get non nan years"
+    years = [x for x in dfc_imf_dot.columns if bool(re.match(r"^\d{4}$", x))]
+    if counterpart_country == "All":
         return int(min(years)), int(max(years))
     non_nan_years = (
-        dfc_imf_dot
-        .query(f"`Country Name` == '{country}'")
+        dfc_imf_dot.query(f"`Country Name` == '{country}'")
         .query(f"`Counterpart Country Name` == '{counterpart_country}'")
         .loc[:, years]
-        .dropna(axis=1, how='all')
+        .dropna(axis=1, how="all")
         .columns
     )
     min_year = int(non_nan_years.min())
     max_year = int(non_nan_years.max())
     return min_year, max_year
 
 
+def render_import_export_time_series(economies, country, selected_years, trade_type):
+    """
+    Plot import export time series using altair
+    """
+    min_selected_year, max_selected_year = selected_years
+    top_n_counters = get_top_n_counterpart_countries(
+        economies, country, 5, max_selected_year, trade_type
+    )
+
+    df = dfc_imf_dot[
+        (dfc_imf_dot["Country Name"] == country)
+        & (dfc_imf_dot["Counterpart Country Name"].isin(top_n_counters))
+    ]
+    num_columns = [
```

Comment:
this is very hard to read. I suggest you to wrap the conditions into a function called sth like `is_in_range`. Then you can simplify this to `[col for col in df.columns if is_in_range(col)]`